### PR TITLE
Setticketsvotebits

### PR DESCRIPTION
**Pull request by chappjc**
For the dcrjson addition: https://github.com/decred/dcrd/commit/cc25399f6a78b09957d82e2c98960b24685e1475
And the dcrwallet addition:
https://github.com/decred/dcrwallet/commit/781468da7200558ebed249e32252736e7ea4c880

Needs `EncodeConcatenatedHashed` in dcrd PR https://github.com/decred/dcrd/pull/429

### CODE REVIEW

**Comment by alexlyp** [comment]
are you going to resubmit your PR with fixed EncodeConcatenatedHashes ? I can get this is in if it's ready.

**Comment by chappjc** [comment]
I'm swamped. Sorry

**Comment by alexlyp** [comment]
Oh no worries, wasn't trying to nag :)

**Comment by chappjc** [comment]
EncodeConcatenatedHashes actually implemented, but this PR is still untested.

**Comment by chappjc** [comment]
In dcrrpcclient, `GetTicketsVoteBits`/`GetTicketsVoteBitsAsync` was written with a `[]*chainhash.Hash` so the same was done in this PR for `SetTicketsVoteBits*`.  However, `EncodeConcatenatedHashes` (just merged into dcrd) takes a `[]chainhash.Hash` instead of a `[]*chainhash.Hash` because the output of the existing Decode... was a`[]chainhash.Hash`.  Consistency is achieved within the packages, but dcrjson and dcrrpcclient don't play nice.

Unless I'm missing a trick, the slice of hash pointers input to SetTickets... needs to be converted to a slice of hashes.  That's not much burden on dcrrpcclient, but it's an extra step I wanted to point out.  The alternative is to let `SetTicketsVoteBits*` in this PR take a []chainhash.Hash although `GetTicketsVoteBits*` does not.  Thoughts about this?

Also, it's curious that getTicketsVoteBits in dcrwallet/rpc/legacyrpc takes hashes as a []string instead of concatenated into a single string as setTicketsVoteBits does.  No decoding necessary there.

File: wallet.go [hunk 1]
```diff
@@ -3327,6 +3328,42 @@ func (c *Client) SetTicketVoteBits(hash *chainhash.Hash, voteBits uint16) error
...
 
+// FutureSetTicketsVoteBitsResult is a future promise to deliver the result of a
+// SetTicketsVoteBitsAsync RPC invocation (or an applicable error).
+type FutureSetTicketsVoteBitsResult chan *response
+
+// Receive waits for the response promised by the future and returns the info
+// provided by the server.
+func (r FutureSetTicketsVoteBitsResult) Receive() error {
+	_, err := receiveFuture(r)
+	return err
```
Comment (chappjc) [inline wallet.go:3339]: Any reason to use previous?

``` go
if err != nil {
    return err
}

return nil
```

Comment (alexlyp) [reply wallet.go:3339]: I suppose for readability, but this way is acceptable.

**Review by alexlyp — APPROVED** [review]
@jrick has talked to me about trying to make an effort to change to []Hashes instead of []*Hashes when possible.  So this would keep in line with the direction we want to go.

OK

**Pull request merged**